extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -905,10 +914,13 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
       return;
     }
 
-    const referenceDisplaySet = displaySetService.getDisplaySetByUID(
-      segOrRTSOverlayDisplaySet.referencedDisplaySetInstanceUID
-    );
-    const imageIds = referenceDisplaySet.images.map(image => image.imageId);
+    let imageIds;
```

Comment:
Please extract the segOrRTSOverlayDIsplaySet logic into it's own function on line 902 up above, to use the same logic as requested in the last comment.